demos/magic_squares_game_optimization.ipynb

```diff
@@ -99,21 +99,13 @@
    "id": "a3377569",
    "metadata": {},
    "outputs": [
-    {
-     "name": "stderr",
-     "output_type": "stream",
-     "text": [
-      "/Users/brian/opt/anaconda3/envs/qnetvo-dev/lib/python3.8/site-packages/pennylane/_grad.py:95: UserWarning: Starting with PennyLane v0.21.0, when using Autograd, inputs have to explicitly specify requires_grad=True (or the argnum argument must be passed) in order for trainable parameters to be identified.\n",
-      "  warnings.warn(\n"
-     ]
-    },
     {
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "opt score :  0.8888798713795558\n",
-      "CPU times: user 30.6 s, sys: 102 ms, total: 30.7 s\n",
-      "Wall time: 30.7 s\n"
+      "opt score :  0.8888798713795557\n",
+      "CPU times: user 42.5 s, sys: 183 ms, total: 42.6 s\n",
+      "Wall time: 42.9 s\n"
```

Comment:
Wonder if this is indicative of a slowdown, or maybe you were running on a different machine/laptop unplugged?

Reply:
Unfortunately, I was able to reproduce this slow down on a number of examples. The gradient descent optimization takes roughly 33% longer in v0.2 than it did in the previous update using pennylane 0.23.

My guess is that the performance problem might be caused by storing the settings as individual tensors withe requires grad set independently for each element. Otherwise, has pennylane experienced a general slowdown over the past few releases?